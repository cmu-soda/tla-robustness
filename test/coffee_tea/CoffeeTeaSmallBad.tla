--------------------------- MODULE CoffeeTeaSmallBad ---------------------------

EXTENDS Naturals

VARIABLES brewed, temp

vars == <<brewed, temp>>

Init == brewed = {} /\ temp = 70

BrewCoffee ==
    /\ brewed = {}
    /\ brewed' = {"coffee"}
    /\ temp' = 212

BrewTea ==
    /\ brewed = {}
    /\ brewed' = {"tea"}
    /\ temp' \in {212, 225, 230}

BrewHotChocolate ==
    /\ brewed = {}
    /\ brewed' = {"hot_chocolate"}
    /\ temp' \in {212, 225}

Next ==
    \/ BrewCoffee
    \/ BrewTea
    \/ BrewHotChocolate
    
Spec == Init /\ [][Next]_vars

WaterSafeTemp == temp <= 220
WaterSafeTempProp == [][WaterSafeTemp]_vars

TypeOK ==
    /\ brewed \in SUBSET {"coffee", "tea", "hot_chocolate"}
    /\ temp \in Nat

BrewOneItem == \A i1,i2 \in brewed : i1 = i2

OnlyErrors == [][~WaterSafeTemp]_vars
OnlyErrorsInv == ~WaterSafeTemp

=============================================================================
\* Modification History
\* Last modified Thu Feb 09 16:05:27 EST 2023 by idardik
\* Created Thu Feb 09 15:03:34 EST 2023 by idardik

