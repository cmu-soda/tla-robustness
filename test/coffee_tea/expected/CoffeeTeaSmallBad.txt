TLA+ Module: CoffeeTeaSmallBad
Diff rep grouped by action:

BrewTea:
Safety boundary representation: out/CoffeeTeaSmallBad_BrewTea_diff_rep.txt
Safety boundary rep formula (with sort definitions):
/\ brewed = {}
/\ temp = 70

BrewHotChocolate:
Safety boundary representation: out/CoffeeTeaSmallBad_BrewHotChocolate_diff_rep.txt
Safety boundary rep formula (with sort definitions):
/\ brewed = {}
/\ temp = 70
